src/components/WorkspaceAdd/WorkspaceAddButton.tsx

```diff
@@ -65,7 +65,11 @@ const WorkspaceAddButton = ({ workspaces }: { workspaces: K8sResourceCommon[] })
     })
       .then((response) => {
         setIsModalOpen(!isModalOpen);
-        navigate(`${response.metadata.name}`);
+        if (response?.metadata?.name) {
+          navigate(`${response.metadata.name}`);
+        } else {
+          throw new Error('Workspace successfully added. Unable to direct to details page.');
```

Comment:
@kdoberst are we guaranteed that the workspace was successfully added in this case? I think it might be cleared to just throw an error indicating we were unable to redirect/navigate.

Reply:
We are sure that the backend returned a successful status code. If it didn't then it wouldn't get to line 71 - it would immediately go to the catch block.

Line 71 is a really edge case where the api returned success but, for whatever reason, didn't pass a name back.